app/controllers/rsvps_controller.rb

```diff
@@ -0,0 +1,37 @@
+class RsvpsController < ApplicationController
+
+  def edit
+    @party = Party.find(params[:party_id])
+    current_user_guest_list_check(@party)
+    @rsvp = Rsvp.find(params[:id])
+    if @rsvp.user_id != current_user.id
+      flash[:notice] = "Sorry, you don't have permission to see that."
+      redirect_to user_path(current_user)
```

Comment:
If you got rid of current_user_guest_list_check you could have all the ownership checks in one place here. Also, it's totally legitimate to do Rsvp.find_by(id: params[:id], user_id: current_user.id) in one go.